Weather/Weather/Scenes/Base/BaseViewController.swift

```diff
@@ -29,14 +30,15 @@ class BaseViewController: UIViewController {
     
     func configureSubview() { }
     
-    @discardableResult
-    func push<T: BaseViewController>(_ viewController: T.Type) -> T? {
-        guard let navigationController = navigationController else {
-            return nil
-        }
-        let controller = viewController.instantiate()
-        navigationController.pushViewController(controller, animated: true)
-        return controller
+    func push<T: BaseViewController>(_ viewController: T) {
```

Comment:
```suggestion
    func push<T: BaseViewController>(_ viewController: T, withAnimation animate: Bool = true) {
```
tăng tính linh hoạt của code